server/sonar-server/src/main/java/org/sonar/server/user/ws/SearchAction.java

```diff
@@ -137,7 +144,7 @@ private User towsUser(UserDto user, @Nullable Integer tokensCount, Collection<St
       .setLogin(user.getLogin());
     setIfNeeded(FIELD_NAME, fields, user.getName(), userBuilder::setName);
     if (userSession.isLoggedIn()) {
-      setIfNeeded(FIELD_EMAIL, fields, user.getEmail(), userBuilder::setEmail);
+      setIfNeeded(FIELD_AVATAR, fields, user.getEmail(), u -> userBuilder.setAvatar(avatarResolver.create(user)));
```

Comment:
Too complex because of the "f" usage, but setIfNeeded is handling nullable value